### DISCUSSION TITLE
Licensing

### DISCUSSION THREAD

**Discussion by lgalabru** [stacks-network/stacks-core#2544] [General]
I'd love to understand our rationale and vision when it comes to licensing. More specifically:
Why going with `GPL 3.0` for our core repo?
Why going with `BSD 2 Clause` for our SIPs?

Also, per my understanding, I think we're supposed to set a license for the Clarity language specification. Since Clarity is defined in a SIP, does it means that its license is `BSD 2 Clause`?
Thanks!

**Comment by jcnelson** [comment]
> Why going with GPL 3.0 for our core repo?

This keeps the "code parasites" away.  We've been approached by large corporations in the past who wanted to create proprietary white-labeled versions of our software to sell to their customers, and the GPLv3 stopped them from doing this.  They weren't even willing to pay a dual-licensing fee to get a one-off non-transferable license to white-label, so as far as I can see here, the GPLv3 protected us from this blatant opportunism.

> Why going with BSD 2 Clause for our SIPs?

They're not code, so there's no money to be made by making proprietary derivative works.  Also, we're not limited to using BSD 2-Clause; SIP-000 has the full list of acceptable licenses for SIPs.

> Also, per my understanding, I think we're supposed to set a license for the Clarity language specification. Since Clarity is defined in a SIP, does it means that its license is BSD 2 Clause?

The SIPs that currently specify Clarity are BSD 2-Clause, but the Foundation holds their copyrights.  So, the Foundation can later change the licensing to something more restrictive (or more open) if need be.  It's up to whoever writes the SIPs what the license will be.
